x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/original_scenarios.md

```diff
@@ -0,0 +1,264 @@
+
+```Gherkin
+Given user navigates to Rules Management Page
+And user is running a fresh instance
+And rule package installation fails
+Then no callouts message should be displayed
+And the number of available rules to install and upgrade in the badges does not change
+```
+
+### Add Elastic Rules Page
+
+#### **Scenario: New workflow elements are displayed on Rules Management Page**
+
+```Gherkin
+Given a user that doesn’t have `security_detection_engine` package installed
+When user is on Rules Management Page
+Then "+Add Elastic rules" menu with available Rules counter is displayed
+And Rule Updates tab is displayed
+And "+Add Elastic rules" button is displayed on empty Rules Table
+```
+
+#### **Scenario: Rules settings persist on Add Elastic Rules table**
+
+```Gherkin
+Given a user has Rules listed on Add Elastic Rules page
+When <case>
+Then the available rules state should persist for all the rules
+
+Examples:
+  | case                              |
+  | user reloads the page             |
+  | after switching table pagination  |
+  | after filtering and clear filters |
+```
+
+#### **Scenario: User can navigate back to Rules Management page**
+
+```Gherkin
+Given a user is on Add Rules Page
+When user navigates back to Rules Management page
+Then Rules Management Page is properly displayed
+```
+
+#### **Scenario: User can filter prebuilt rules by rule name or by tag**
+
+```Gherkin
+Given a user is on Add Rules Page
+When user filters by <filter>
+Then Add Rules Table is properly updated
+
+Examples:
+  | filter                  |
+  | rule name on search bar |
+  | Tag filter              |
+```
+
+### Rule Updates tab
+
+#### **Scenario: Rules settings persist on Rule Updates table**
+
+```Gherkin
+Given a user has Rules listed on Rule Updates table
+When <case>
+Then the rules with available updates state should persist
+
+Examples:
+  | case                              |
+  | user reloads the page             |
+  | after switching table pagination  |
+  | after filtering and clear filters |
+```
+
+#### **Scenario: User can navigate back to Rules Management tab**
+
+```Gherkin
+Given a user is on Rule Updates tab
+When user navigates back to Rules Management page
+Then Rules Management Page is properly displayed
+```
+
+#### **Scenario: User can filter prebuilt rules by rule name or by tag**
+
+```Gherkin
+Given a user is on Rule Updates tab
+When user filters by <filter>
+Then Rule Updates tab is properly updated
+
+Examples:
+  | filter                  |
+  | rule name on search bar |
+  | Tag filter              |
+```
+
+### Authorization / RBAC
+
+#### **Scenario: User with read privileges on security solution cannot install prebuilt rules**
+
+```Gherkin
+Given a user with Security: read privileges on Security solution
+When user navigates to Add Elastic Rules Page
+Then user can see available prebuilt rules to install
+And user cannot Install those prebuilt rules
+```
```

Comment:
Replaced by:

#### **Scenario: User with read privileges on Security Solution cannot install prebuilt rules**

**Coverage**: 1 e2e test with mock rules + 3 integration tests with mock rules for the status and installation endpoints.

```Gherkin
Given user with "Security: read" privileges on Security Solution
And no prebuilt rules are installed in Kibana
And there are prebuilt rules available to install
When user opens the Add Rules page
Then user should see prebuilt rules available to install
But user should not be able to install them
```